### ISSUE TITLE
Temporary files growing and filling disk 

### ISSUE THREAD

**Issue by betonetotbo** [classgraph/classgraph#239]
Why Fast Classpath Scanner are creating this jars and letting them on my disk?

![image](https://user-images.githubusercontent.com/5444923/44586684-facd1280-a786-11e8-8a98-a8cb35894f08.png)

This is about *3.5 GB* of unnecessary files!

In another customer I found more than **900GB**.

*Library:* fast-classpath-scanner
*Versão:* 2.2.1

**Comment by lukehutch** [comment]
You're using a very old version of the library! The newer version (Classgraph 4.0.7) does a better job of removing temporary files. Please port your code to Classgraph, and assign the ScanResult in a try-with-resources block as shown in the docs.

**Comment by lukehutch** [comment]
PS please let me know if this fixes the problem for you.

**Comment by betonetotbo** [comment]
@lukehutch I'll test this new version.

For curiosity, I use this library just to list resources in a custom URLClassLoader (one or more jars that constains jrxmls for jasper reports).

Can you give me some tips how to optimize the ClassGraph configuration to be most efficient possible. Some tips like "disable the nested jars scan" and so...

I just want to configure the scanner to be fast as possible without unecessary overheads.

**Comment by lukehutch** [comment]
For your specific use case, you would call

```
try (ScanResult scanResult = new ClassGraph()
    .overrideClassloaders(customURLClassloader)
    .disableNestedJarScanning()
    .whitelistPathsNonRecursively(jrxmlDir)
    .scan()) {
        Resource List resources = scanResult.getResourcesWithExtension("xml");
    //... 
} 
```

Overriding classloaders with your own means you only scan the jars you care about. Disabling Nested jar scanning you already know. Whitelisting paths non-recursively means you don't scan anything but the listed path(s).

Collectively that's about as fast as you can scan resources only.

There might be some typos in the method names above, I'm not in front of a computer right now :)

**Comment by lukehutch** [comment]
@betonetotbo I finally figured out why temporary files were not being deleted sometimes in Windows. Windows cannot delete a file that is mmap'd, and an mmap'd file can only be un-mmap'd through garbage collection (#288). This should be fixed in the 4.6.4 release of ClassGraph. (Also 4.6.x generates significantly fewer temporary files, since non-deflated/stored nested jars no longer need to be extracted to temporary files.) Sorry that it took so long to figure this out!
